cicecore/cicedynB/infrastructure/ice_grid.F90

```diff
@@ -384,7 +384,7 @@ subroutine init_grid2
       ! T-grid cell and U-grid cell quantities
       !-----------------------------------------------------------------
 
-!     tarea(:,:,:) = c0
+      tarea(:,:,:) = c0
```

Comment:
I think this is unnecessary as tarea is in a HaloUpdate shortly after it is computed. I don't think you want any zero values in tarea at all.

Reply:
Thanks for the review. I agree that it should not be necessary, but as I wrote, if I do not initialize here, an uninitialized value gets used... So that would mean that not all halo cells are initialized by the halo update. I'll have to investigate that further on the specific grid we are using.

Reply:
It could be that your compiler is being very picky. What line does it abort on exactly with the uninitialized variable? I've done the check with NAG and Intel and it runs fine. As you say, it might be a particular feature of your grid. There was an issue awhile ago from one of the European groups that trapped this same error. I am fine with initializing uvm to zero, but tarea should never be zero.

Reply:
It aborts at line [1673 in `makemask`](https://github.com/CICE-Consortium/CICE/blob/a76893a0d47c3c2d0c8fbe83238aa3b000dabd1b/cicecore/cicedynB/infrastructure/ice_grid.F90#L1694)

Reply:
Ok. A cleaner initialization would be as follows:

do j = 1, ny_block
   do i = 1, nx_block

      im = max(i,ilo)
      jm = max(j,jlo)
      ip = min(i,ihi)
      jp = min(j,jhi)

Then replace all of the i,j indexes on the right with ip/jp. This will extrapolate the tarea values to the ghost cells including the four corners. Thoughts?

Reply:
Sorry, these should all be im/jm or ip/jp.

Reply:
If the maskhalo is turned on, it's possible that is also masking some part of the initialization.  Just speculation on my part, we'd have to look into it further.

Reply:
Actually, the maskhalo is not being used for the halo update of tarea, which is correct, so forget I said that.   I assume tarea is not initialized on the entire halo because of the land mask?  We could also fix the loop in makemask, but that's likely to affect other loops in other parts of the code in the same way.  This is almost certainly a case of variables not being initialized in places they are not needed, but where they are referenced at some point later.   The proper fix, of consistently implementing all loops, is probably not worth the effort, but if we want to update a loop here or there, that's probably OK too.  I'm not opposed to this type of fix in this case.